src/chromo/common.py

```diff
@@ -438,14 +444,19 @@ def to_hepmc3(self, genevent=None):
             ev = self[mask]
             mask = (ev.mothers[:, 0] == 0) | (ev.mothers[:, 0] == 1)
             ev.mothers[mask] = (0, 1)
-        elif model in ("UrQMD", "PhoJet", "DPMJET-III"):
-            # can only save final state until history is fixed
-            warnings.warn(
-                f"{model}-{version}: only final state particles "
-                "available in HepMC3 event",
-                RuntimeWarning,
+        elif model == "PhoJet":
```

Comment:
The specific of the models should be handled by the model-related classes and not in common. Common should remain model agnostic as much as possible.

Reply:
Move the code to the corresponding classes